src/RA_LeaderboardPopup.cpp

```diff
@@ -273,9 +276,9 @@ _Use_decl_annotations_ void LeaderboardPopup::Render(ra::ui::drawing::ISurface&
             if (m_pScoreboardSurface != nullptr)
             {
                 const auto fOffscreenAmount = (GetOffsetPct() * 600);
-                const auto fFadeOffs = (nWidth - 300 - 10) + fOffscreenAmount;
-                const int nScoreboardX = static_cast<int>(fFadeOffs);
-                const int nScoreboardY = nHeight - 200 - 10;
+                const auto fFadeOffs = (gsl::narrow<std::int64_t>(nWidth - 300 - 10) + fOffscreenAmount);
+                const auto nScoreboardX = ra::ftoi(fFadeOffs);
```

Comment:
what does the `narrow` do here? it looks like it's trying to convert the float to an int64, which implies the `ftoi` on the next line is unnecessary.

Reply:
It's for narrowing conversions. Since double is 8 bytes, and nwidth is 4, it needs to be cast yo a wider type to prevent overflow

Reply:
The code wont compile if it wasn't necessary as the function  requires a floating point number at compile time

Reply:
The int needs to be cast to a wider type to prevent overflow.
This warning comes up:
`Warning	C26451 Arithmetic overflow: Using operator '-' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '-' to avoid overflow (io.2)`

_The rule ids are not the same because "io" is used for `SL.io: Iostream` in the guidelines._
Here's what it says in the C++ Core Guidelines (linked from MS Docs)
```
ES.103: Don't overflow
Reason
Overflow usually makes your numeric algorithm meaningless. Incrementing a value beyond a maximum value can lead to memory corruption and undefined behavior.
```

Reply:
I dont know. Its complained before about multiplying and division (they changed since then do wont show up now). I'm pretty sure its complaint about one operand having a smaller byte width which makes to me. It doesnt seem immediately apparent unless it's with binary format.

I will try your suggestion anyway.

Reply:
Your suggestion worked, I'm going to keep that in mind for the future. I also added the Arithmetic rules for CppCoreCheck since it's completely including in this one. Mainly so we don't have to run it.